pstore/api.go

```diff
@@ -34,11 +37,108 @@ type Record struct {
 	Timestamp time.Time
 }
 
+// Writer is the interface for writing to persistent store.
+// Implementations of Writer must be safe to use with multiple goroutines.
 type Writer interface {
+	// Write writes given collection of records to persistent storage
+	Write(records []Record) error
+}
```

Comment:
Pstorewriter?

On Friday, April 22, 2016, rgooch <notifications@github.com> wrote:

> In pstore/api.go
> <https://github.com/Symantec/scotty/pull/51#discussion_r60773202>:
>
> >  type Writer interface {
> > +	// Write writes given collection of records to persistent storage
> > +	Write(records []Record) error
> > +}
>
> I think it may be confusing to re-use the standard Writer interface name
> for a different interface. I have a preconception that "everybody knows"
> that a Writer interface is io.Writer. What about PStoreWriter or
> RecordWriter or something?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Symantec/scotty/pull/51/files/4bbff0e1806a654cfa1c3380775d6ddbbcbfd275#r60773202>
>